airqo-platform/AirQo-api
# Join Platform (forgot password)

- What does this PR do?
This PR creates 3 new endpoints of three new endpoints: updatePasswordViaEmail, reset password, and forgotPassword. Addresses security concerns and implements additional access control checks.
- What JIRA task is related to this PR?
The links to the related JIRA boards task are [1](https://airqoteam.atlassian.net/browse/LOC-16) and [2](https://airqoteam.atlassian.net/browse/NLYTCS-1)
- How do I test out this PR?
Git clone this repo `cd AirQo-api/src/auth-service` Then run `npm install` to get all the necessary dependencies Afterwards run `npm run test` to test out the end points using any REST client tool available to you.
- Which end points should are ready for testing?
[AirQo RESTful API documentation](https://docs.google.com/spreadsheets/d/1-NGTDBtTcIi8Fg2UYSWjcK6geKrZdopz5kxRFsCEMPs/edit?usp=sharing)

1. partial url: /api/v1/users/updatePasswordViaEmail
2. partial url: /api/v1/users/reset?<TOKEN>
3. partial url: /api/v1/users/forgotPassword/<EMAIL>
4. partial url: /api/v1/users/email/confirm/<user ID>
5. partial url: /api/v1/users/register/collab
6. partial url: /api/v1/users/update/collab/<user ID>
7. partial url: /api/v1/users/update/collab/<user ID>
8. partial url: /api/v1/users/updatePasswordViaEmail
